docs/upgrade/index.rst

```diff
@@ -6,10 +6,10 @@ Upgrade ScyllaDB
    :titlesonly:
    :hidden:
 
-   ScyllaDB Enterprise <upgrade-enterprise/index>
    ScyllaDB Open Source <upgrade-opensource/index>
    ScyllaDB Open Source to ScyllaDB Enterprise <upgrade-to-enterprise/index>
    ScyllaDB AMI <ami-upgrade>
+   ScyllaDB Enterprise <https://enterprise.docs.scylladb.com/enterprise/upgrade/index.html>
```

Comment:
Why not pointing directly to
https://enterprise.docs.scylladb.com/enterprise/upgrade/upgrade-enterprise/index.html

Reply:
The overview page includes general information about the upgrade and an example (now it's an example for Open Source, but in the end, we'll have another one for Enterprise. I thought it would be useful for the user to read it and then they can choose a subpage they are interested in (Enterprise->Enterprise, OSS->Enterprise, AMI).
I can update the link if you think it's more useful to link directly to the Enterprise->Enterprise upgrades, I'll update the link.

Reply:
@tzach Updated in one commit. Please re-review.